### PR TITLE
ENH: Creates a config.yaml file to hold badger's settings 

### PR DESCRIPTION
**Pull request by YektaY**
## Description
This PR creates a config.yaml file to hold badger's settings. it also changed the internal plumbing of the settings. Changing from a design which used global variables paired with QSettings to a design which uses pydantic class which holds the default settings, a config file which badger loads in to replace the default settings and a singleton to hold the settings and pipe it throughout the rest of the application. 

## How Has This Been Tested?
Unit tests need to be added to make this PR ready for release 

## Where Has This Been Documented?
Auto documentation has been added to all new methods. 
The process will be noted on the website 

## Screenshots:

### CODE REVIEW

File: src/badger/config_template.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+BADGER_ARCHIVE_ROOT: /Users/yazar/Library/Application Support/Badger/archive
```
Comment (YektaY) [inline src/badger/config_template.yaml:1]: whoops, this file will be removed

File: src/badger/tests/test_routine_page.py [hunk 2]
```diff
@@ -96,8 +96,8 @@ def test_add_additional_vars(qtbot):
     assert window.env_box.var_table.rowCount() == n_rows + 1
 
     # Check VOCS
```
Comment (roussel-ryan) [inline src/badger/tests/test_routine_page.py:98]: should be removed if not needed

**Review by roussel-ryan — COMMENTED** [review]
Looks good to me, need to remove some commented out lines if we don't need to

**Comment by roussel-ryan** [comment]
Also need to resolve merge conflicts

**Comment by YektaY** [comment]
I removed the commented out code and fixed the merge conflict. let me know if there is anything else needed on this PR

**Review by roussel-ryan — APPROVED** [review]

**Pull request merged**